internal/pkg/service/buffer/storage/compression/writer/writer.go

```diff
@@ -12,10 +12,10 @@ import (
 )
 
 // New wraps the specified writer with the compression writer.
-func New(w io.Writer, cfg compression.Config) (io.Writer, error) {
+func New(w io.Writer, cfg compression.Config) (io.WriteCloser, error) {
 	switch cfg.Type {
 	case compression.TypeNone:
-		return w, nil
+		return &nopCloser{Writer: w}, nil
```

Comment:
neexistuje ekvivalent `io.NopCloser` pre writer, je to v standardnej kniznici iba pre reader.